Bottom: a5f9c12111369a5fe835c02aa3f6321f8f124984
Top:    3af74542971ae63ba8cb3ceccf9c3c28bd3efaec
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-18 21:22:39 +0100

Refresh of initialise-the-hid-vtable-by-f

---

diff --git a/src/hid/common/extents.c b/src/hid/common/extents.c
index 16bd75a..49f681e 100644
--- a/src/hid/common/extents.c
+++ b/src/hid/common/extents.c
@@ -28,7 +28,6 @@ RCSID ("$Id$");
 
 static BoxType box;
 
-
 typedef struct hid_gc_struct
 {
   int width;
@@ -171,9 +170,10 @@ extents_fill_rect (hidGC gc, int x1, int y1, int x2, int y2)
 }
 
 static HID extents_hid;
+static bool initialised = false;
 
 void
-init_extents_hid (void)
+hid_extents_init (void)
 {
   memset (&extents_hid, 0, sizeof (HID));
 
@@ -204,6 +204,15 @@ init_extents_hid (void)
 BoxType *
 hid_get_extents (void *item)
 {
+  /* As the extents "hid" isn't a real HID, we need to ensure it is
+   * initialised before use from every external entry point.
+   */
+  if (!initialised)
+    {
+      hid_extents_init ();
+      initialised = true;
+    }
+
   BoxType region;
 
   box.X1 = MAXINT;
